cogeo_mosaic/backends/dynamodb.py

```diff
@@ -23,22 +23,67 @@
 from cogeo_mosaic.mosaic import MosaicJSON
 from cogeo_mosaic.utils import bbox_union
 
+# From https://docs.aws.amazon.com/general/latest/gr/rande.html
+AWS_REGIONS = [
+    "af-south-1",
+    "ap-east-1",
+    "ap-northeast-1",
+    "ap-northeast-2",
+    "ap-northeast-3",
+    "ap-south-1",
+    "ap-southeast-1",
+    "ap-southeast-2",
+    "ca-central-1",
+    "cn-north-1",
+    "cn-northwest-1",
+    "eu-central-1",
+    "eu-north-1",
+    "eu-south-1",
+    "eu-west-1",
+    "eu-west-2",
+    "eu-west-3",
+    "me-south-1",
+    "sa-east-1",
+    "us-east-1",
+    "us-east-2",
+    "us-west-1",
+    "us-west-2",
+]
+
 
 @attr.s
 class DynamoDBBackend(BaseBackend):
     """DynamoDB Backend Adapter."""
 
     client: Any = attr.ib(default=None)
-    region: str = attr.ib(default=os.getenv("AWS_REGION", "us-east-1"))
-    table_name: str = attr.ib(init=False)
+    region: str = attr.ib(
+        default=os.getenv("AWS_REGION", "us-east-1"),
+        validator=attr.validators.in_(AWS_REGIONS),
+    )
+    table_name: str = attr.ib(
+        init=False, validator=attr.validators.matches_re(r"^[a-zA-Z0-9\_\-\.]+$")
```

Comment:
@kylebarron validators only run on init, so this won't work

Reply:
@kylebarron because the `table_name` is set during the `__post_init` the validation is not happening! you'll need to re-run the validation at the end of the `__post_init__`